source/files/compression/compression.c

```diff
@@ -0,0 +1,257 @@
+
+    if(!binaries)
+        return 1;
+
+    FILE* output = NULL;
+    output = fopen(outputPath, "w");
+
+    char ch = '\0';
+    Tree* searcher = arbre;
+
+    while ((ch = fgetc(binaries)) != EOF)
+    {
+        if (!hasSons(searcher))
+        {
+            fputc(searcher->data, output);
+            searcher = arbre;
+        }
+
+        if(ch == '0'){
+            searcher = searcher->left;
+        }else{
+            searcher = searcher->right;
+        }
+    }
+
+    if (!hasSons(searcher))
+    {
+        fputc(searcher->data, output);
+        searcher = arbre;
+    }
+
+    fclose(binaries);
+    fclose(output);
+    return 0;
+}
+
+void b_decompression(char *target, Tree *tree)
+{
+    FILE* binaries = NULL;
+    binaries = fopen(target, "rb");
+
+    if(!binaries)
+        return;
+
+    FILE* output = NULL;
+    output = fopen("output.txt", "w");
+
+    char ch = '\0';
+    Tree* searcher = tree;
+
+    // overbytes size
+    fread(&ch, sizeof(char), 1, binaries);
+    int bsize = (int)ch;
+    ch = '\0';
+
+    //get file size
+    int f_size = get_file_size(binaries);
+
+    // iterator
+    int i = 0;
+
+    while (ch != EOF && i < (f_size - bsize))
+    {
+        fread(&ch, sizeof(char), 1, binaries);
+        i++;
+
+        if (!hasSons(searcher))
+        {
+            //printf("%c", searcher->data);
+            fputc(searcher->data, output);
+            searcher = tree;
+        }
+
+        if(ch){
+            searcher = searcher->right;
+        }else{
+            searcher = searcher->left;
+        }
+    }
+
+    if (!hasSons(searcher))
+    {
+        fputc(searcher->data, output);
+        searcher = tree;
+    }
+
+    fclose(binaries);
+    fclose(output);
+}
+
+int b_compression(char *target)
+{
+    // --- file read ---
+    // file variables
+    FILE* input = fopen(target, "r"); if (!input) return 1;
+    int size = get_file_size(input)+1;
+
+    // text variables
+    char* transcription = (char*)malloc(size*sizeof(char));
+    char ch = '\0';
+
+    // iterator
+    int i = 0;
+
+    // character list
+    Tree* occurrences = NULL;
+
+    //file read
+    while ((ch = fgetc(input)) != EOF)
+    {
+        //printf("%c : %d \n", ch, ch);
+
+        // update transcription
+        transcription[i] = ch;
+        i++;
+
+        // update list
+        add_occurrences(&occurrences, ch);
+    }
+
+    // end the file transcription
+    transcription[i] = '\0';
+
+    // close file
+    fclose(input);
+
+    // list to dictionary
+    Element* HS = BT_to_UT(occurrences);
+    SLL_to_HT(&HS);
+    Tree* HT = HS->node;
+    Dico* dictionary = htreetodico(HT, occurrences);
+
+    // temporary
+    //Tree* h_tree = sort_SLL_to_BT(list);
+    // free SLL
+    free_SLL(HS);
+
+    // --- compression ---
+
+    /* Available Variables
+     * Dico Dictionary
+     * int size
+     * char* transcription
+     * char ch
+     * int i
+     * */
+
+    char* outputPath = (char*)malloc(300*sizeof(char));
```

Comment:
2 choices :
- char outputPath[300] (no need for a dynamic allocation, so why not a static one)
- put it after the int strSize and allocate strSize space